### ISSUE TITLE
webgl canvas resize will crop result to smaller image then requested

### ISSUE THREAD

**Issue by verliba** [electron/electron#10284]
Hi,
I encountered this strange behaviour in webgl rendering..
when I resize canvas (I think no matter if new size is bigger or smaller), and later I get blob of it, then the blob contains cropped image.
Following code is what I have to do before getting cropped blob..

```
 let canvas = document.createElement('canvas');
        canvas.width=8000; canvas.height=8000;
        let attributes = { premultipliedAlpha: true };
        let gl = canvas.getContext('webgl',attributes);
        canvas.width=6000;  canvas.height=6000;
        gl.viewport(100, 100, 5800, 5800);
        gl.clearColor(0, 0.5, 0, 1);
        gl.clear(gl.COLOR_BUFFER_BIT);
        gl.finish();
        let resBlob = await new Promise(resolve => canvas.toBlob(resolve));
```

There are two flavours of this behaviour, if the premultipliedAlpha is false, the resBlob image has dimensions 4096x4096; when it is true, the blob image dimensions are correct, but the image has transparent areas, and there is 4096x4096 area, that is displayed properly..

4096x4096 seems to be there on purpose, because of number of pixels fits right under 16M.. Different image proportions result in same upper limit of number of pixels.

I hope this is enough info to get it fixed.

**Comment by verliba** [comment]
I expect this is probably more a chromium bug, so I reported it there https://bugs.chromium.org/p/chromium/issues/detail?id=756347

**Comment by sofianguy** [comment]
We are no longer providing bug fixes for bug reports later than 2.0.x. Is this still a bug in later versions of Electron?

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
